feat(rust): Add `crates` output support to generator configuration

iamnamananand996 opened the pull request:
## Description

Introduces support for a new 'crates' output location in generator configuration schemas and serialization. Implements a temporary workaround by mapping crates to npm output handling, allowing crates configuration to pass validation and be processed until native support is added.

```bash
rust-sdk:
    generators:
      - name: fernapi/fern-rust-sdk
        version: 0.0.3
        ir-version: v58
        github:
          repository: fern-demo/foxglove-rust-sdk
          mode: pull-request
        output:
          location: crates
          package-name: foxglove
        config:
          crate_name: foxglove

```

will support 

```bash
location: crates
```


## Changes Made

- Added `CratesOutputLocationSchema` type definition in both API and serialization schemas
- Updated `GeneratorOutputSchema` to include crates as a supported output location type
- Implemented temporary workaround in `convertGeneratorsConfiguration.ts` that maps crates output to npm handling
- Added proper type exports for the new crates schema in index files
- Extended generator configuration validation to accept `crates` output configurations


 ## Testing
  - [x] Manual testing completed

iamnamananand996 committed:
feat(rust): Add crates output support to generator configuration

Introduces support for a new 'crates' output location in generator configuration schemas and serialization. Implements a temporary workaround by mapping crates to npm output handling, allowing crates configuration to pass validation and be processed until native support is added.

dsinghvi reviewed (changes requested):
@iamnamananand996 you need to update https://github.com/fern-api/fern/blob/fe6dd39eec7ac856eb21bb714a40f51d6628470a/fern/apis/generators-yml/definition/group.yml#L54-L63 and then run `pnpm generators:generate` to generate the typings

iamnamananand996 committed:
chore(rust): Add Rust crates publishing configuration support

Introduces configuration schemas for publishing Rust crates to crates.io or custom registries in both YAML and TypeScript definitions. Adds 'crates' as a supported publishing target, with appropriate properties and documentation, and updates field names for consistency.

iamnamananand996 committed:
chore(rust): Rename 'api-key' to 'token' in CratesOutputLocationSchema

Replaces the 'api-key' property with 'token' in the CratesOutputLocationSchema across the YAML definition and TypeScript schema files for consistency and clarity.

iamnamananand996 committed:
chore(rust): add schema file

iamnamananand996 committed:
Merge branch 'main' into naman/support-creates-path-enum
